### PR TITLE
[nRF Connect] Use common server & ZCL code in the examples

### PR DESCRIPTION
**Pull request by Damian-Nordic**
#### Problem
There's quite a lot of duplicated code across examples for different platforms. After merging #2367 we have the opportunity to reuse the common code for ZCL and CHIP application server. 

 #### Summary of Changes
Modify lock-app and lighting-app examples for the nRF Connect platform to use common code extracted to examples/common/ and examples/lock-app/lock-common directories.

### CODE REVIEW

**Review by mspang — APPROVED** [review]

File: examples/lock-app/nrfconnect/main/ZclCallbacks.cpp [hunk 1]
```diff
@@ -1,6 +1,7 @@
 /*
```
Comment (rwalker-apple) [inline examples/lock-app/nrfconnect/main/ZclCallbacks.cpp:1]: does this file differ materially from the one with the same name above?

**Review by rwalker-apple — APPROVED** [review]
much better, thank you

one question above, though: might we get even a bit more common code?

File: examples/lock-app/nrfconnect/main/ZclCallbacks.cpp (hunk 1 above)
Comment (rwalker-apple) [reply examples/lock-app/nrfconnect/main/ZclCallbacks.cpp:1]: nevermind, I see how this differs

**Comment by Damian-Nordic** [comment]
> much better, thank you
> 
> one question above, though: might we get even a bit more common code?

I don't know to be honest. I remember seeing a comment in one of the PRs that there's a plan to create some kind of application component within the CHIP library, but I don't know the plan. Making remaining application pieces more common (e.g. AppTask.cpp) would require a proper abstraction layer over OS primitives like threads, timers etc. which I'm not sure is desired (?). Perhaps CHIP initialization process could be unified though. On the nRF Connect side, we will certainly be working to unify configuration of the examples a bit.

**Review by BroderickCarlin — APPROVED** [review]

**Pull request merged**